pkg/v1/config/clientconfig.go

```diff
@@ -143,6 +146,75 @@ func NewClientConfig() (*configv1alpha1.ClientConfig, error) {
 	return c, nil
 }
 
+func populateDefaultStandaloneDiscovery(c *configv1alpha1.ClientConfig) bool {
+	if c.ClientOptions == nil {
+		c.ClientOptions = &configv1alpha1.ClientOptions{}
+	}
+	if c.ClientOptions.CLI == nil {
+		c.ClientOptions.CLI = &configv1alpha1.CLIOptions{}
+	}
+	if c.ClientOptions.CLI.DiscoverySources == nil {
+		c.ClientOptions.CLI.DiscoverySources = make([]configv1alpha1.PluginDiscovery, 0)
+	}
+
+	switch DefaultStandaloneDiscoveryType {
+	case common.DiscoveryTypeOCI:
+		return populateDefaultStandaloneDiscoveryOCI(c)
+	case common.DiscoveryTypeLocal:
+		return populateDefaultStandaloneDiscoveryLocal(c)
+	default:
+		log.Warning("unsupported default standalone discovery configuration")
+	}
+	return false
+}
+
+func populateDefaultStandaloneDiscoveryLocal(c *configv1alpha1.ClientConfig) bool {
+	for _, ds := range c.ClientOptions.CLI.DiscoverySources {
+		if ds.Local != nil && ds.Local.Name == DefaultStandaloneDiscoveryName {
+			if ds.Local.Path == DefaultStandaloneDiscoveryLocalPath {
+				return false
+			}
+			ds.Local.Path = DefaultStandaloneDiscoveryLocalPath
+			return true
+		}
+	}
+
+	defaultDiscovery := configv1alpha1.PluginDiscovery{
+		Local: &configv1alpha1.LocalDiscovery{
+			Name: DefaultStandaloneDiscoveryName,
+			Path: DefaultStandaloneDiscoveryLocalPath,
+		},
+	}
+
+	// Prepend default discovery to available discovery sources
+	c.ClientOptions.CLI.DiscoverySources = append([]configv1alpha1.PluginDiscovery{defaultDiscovery}, c.ClientOptions.CLI.DiscoverySources...)
+	return true
+}
```

Comment:
This function doesn't seem to be covered by unit tests.

Reply:
added unit test

Reply:
Thanks for adding tests! The `for` loop body is still not covered, and it contains non-trivial logic. Perhaps, it even makes sense to extract it as a separate function.